tests/runtimes/android/android_runtime_tests.py

```diff
@@ -0,0 +1,271 @@
+                   'Successfully synced application']
+
+        test_result = Wait.until(lambda: all(string in File.read(log.log_file) for string in strings), timeout=300,
+                                 period=5)
+        assert test_result, 'Application is not build successfully!'
+        Device.wait_for_text(self.emulator, "TAP")
+        Adb.is_text_visible(self.emulator.id, "TAP", True)
+        Device.click(self.emulator, "TAP", True)
+
+        strings = ["Error: java.lang.Exception: Failed resolving method createTempFile on class java.io.File",
+                   "Caused by: java.lang.Exception: Failed resolving method createTempFile on class java.io.File"]
+
+        test_result = Wait.until(lambda: all(string in File.read(log.log_file) for string in strings), timeout=300,
+                                 period=5)
+        assert test_result, 'Native crash should not crash the app when discardUncaughtJsExceptions used fails!'
+        Device.screen_match(self.emulator,
+                            os.path.join(TEST_RUN_HOME, 'assets', 'runtime', 'android', 'images',
+                                         'Emulator-Api23-Default',
+                                         "No-crash-image.png"), timeout=120, tolerance=1)
+
+    def test_318_generated_classes_not_be_deleted_on_rebuild(self):
+        """
+            https://github.com/NativeScript/nativescript-cli/issues/3560
+        """
+        Tns.platform_remove(app_name=APP_NAME, platform=Platform.ANDROID)
+        Tns.platform_add_android(APP_NAME, framework_path=Android.FRAMEWORK_PATH)
+        target = os.path.join(TEST_RUN_HOME, APP_NAME, 'app')
+        source = os.path.join(TEST_RUN_HOME, 'assets', 'runtime', 'android', 'files', 'android-runtime-904',
+                              'MyActivity.js')
+        File.copy(source, target)
+
+        Tns.build_android(os.path.join(TEST_RUN_HOME, APP_NAME))
+
+        assert File.exists(os.path.join(TEST_RUN_HOME, APP_NAME, "app", "MyActivity.js"))
+        assert File.exists(
+            os.path.join(TEST_RUN_HOME, APP_NAME, "platforms", "android", "app", "src", "main", "java", "com", "tns",
+                         "MyActivity.java"))
+
+        File.delete(os.path.join(TEST_RUN_HOME, APP_NAME, 'app', 'MyActivity.js'))
+
+        Tns.build_android(os.path.join(TEST_RUN_HOME, APP_NAME))
+
+        assert not File.exists(
+            os.path.join(TEST_RUN_HOME, APP_NAME, "platforms", "android", "app", "src", "main", "java", "com", "tns",
+                         "MyActivity.java"))
+
+    def test_350_sgb_fails_generating_custom_activity(self):
+        """
+        Static Binding Generator fails if class has static properties that are used within the class
+        https://github.com/NativeScript/android-runtime/issues/1160
+        """
+        Folder.clean(os.path.join(TEST_RUN_HOME, APP_NAME))
+        Tns.create(app_name=APP_NAME, template=Template.HELLO_WORLD_JS.local_package, update=True)
```

Comment:
Can we use Template.MIN_JS?

Reply:
All tests are based on hello world js template. I would prefer using it.